velox/common/memory/tests/SharedArbitratorTest.cpp

```diff
@@ -384,29 +384,29 @@ DEBUG_ONLY_TEST_F(SharedArbitrationTest, skipNonReclaimableTaskTest) {
         ++taskPausedCount;
       })));
 
+  auto plan = PlanBuilder()
+                  .values(vectors)
+                  .singleAggregation({"c0", "c1"}, {"array_agg(c2)"})
+                  .planNode();
   std::thread spillableThread([&]() {
     const auto spillDirectory = exec::test::TempDirectoryPath::create();
-    AssertQueryBuilder(PlanBuilder()
-                           .values(vectors)
-                           .singleAggregation({"c0", "c1"}, {"array_agg(c2)"})
-                           .planNode())
+    AssertQueryBuilder(plan)
         .queryCtx(queryCtx)
         .spillDirectory(spillDirectory->getPath())
         .copyResults(pool());
   });
 
+  plan = PlanBuilder()
```

Comment:
The race is because of resetting the type resolver after plan builder dtor? Why TypeResolverHook is not per-thread local? It might be easier to use. Thanks!